### ISSUE TITLE
VDC helm chart - IGC & Atlas connectors are no longer aligned with release

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#3415]
When testing the odpi-egeria-vdc helm chart with the 2.1 release branch, I noticed that the egeria connector isn't loading:
```
vdc-odpi-egeria-vdc-atlas-6fb46c5688-2ftsm             0/2     Init:CrashLoopBackOff   7          15m
```
and
```
Connecting to odpi.jfrog.io (34.83.160.0:443)
wget: server returned error: HTTP/1.1 404 Not Found
Connecting to odpi.jfrog.io (34.83.160.0:443)
wget: server returned error: HTTP/1.1 404 Not Found
Connecting to odpi.jfrog.io (34.83.160.0:443)
wget: server returned error: HTTP/1.1 404 Not Found
Connecting to odpi.jfrog.io (34.83.160.0:443)
wget: server returned error: HTTP/1.1 404 Not Found
This is coming from
```
                wget ${EGERIA_ARTIFACTORY_RELEASE}/${ATLAS_CONNECTOR_URL_PATH}/${EGERIA_VERSION}/egeria-connector-apache-atlas-package-${EGERIA_VERSION}-jar-with-dependencies.jar && touch RELEASE;
```
which refers to
```
  EGERIA_ARTIFACTORY_RELEASE: https://odpi.jfrog.io/odpi/egeria-release-local
  EGERIA_ARTIFACTORY_STAGING: https://odpi.jfrog.io/odpi/egeria-staging-local
  EGERIA_ARTIFACTORY_SNAPSHOT: https://odpi.jfrog.io/odpi/egeria-snapshot-local
```
but a search in artifactory to https://odpi.jfrog.io/odpi/webapp/#/search/quick/eyJzZWFyY2giOiJxdWljayIsInF1ZXJ5IjoiZWdlcmlhLWNvbm5lY3Rvci1hcGFjaGUtYXRsYXMtcGFja2FnZSJ9 shows we have no connectors after 1.6-SNAPSHOT

We have a long list of vdc related todos/issues. I am about to close many and just have a placeholder 'what next' though if it's a simple bug fix let's update it. The igc/egeria connectivity is working fine. A trivial fix would be to switch off atlas by default and link back to this issue. I think we're ok with master only since this issue may have been there a while

**Comment by cmgrote** [comment]
I haven't been keeping up with all the Egeria releases in the connectors themselves -- in some cases it just doesn't make sense to even do so as there isn't any connector-relevant change in core (or the connector itself) to require a new connector release.

Probably the better fix is to decouple the version numbers between core and connector(s) (ie. different values var for each)?

**Comment by planetf1** [comment]
Looks as if it's not worked since 1.6 - I haven't tried, but we (I!) could see if the older connector still works and change the chart to use that.

Agree on version decoupling ... plus we have a pending move to another repo. Given no complaints so far, I don't see this as an urgent action?

**Comment by planetf1** [comment]
The last Atlas connector release was 1.6

Also the last IGC connector release was 1.5

It's probably worth seeing if they still work, and can update charts accordingly, but fundamentally we don't currently support the VDC chart well as we've focussed on other things.

I wonder if we should deprecate the chart/update the docs to remind users it's not really what we're focussing on currently (though contributions welcome).

I'll try the simple version update first

**Comment by planetf1** [comment]
I've added two new variables to track the igc & atlas connector releases 

On versions - My bad - my search was truncated. the connectors do have a 2.1 release, so I will leave them set there.

**Comment by cmgrote** [comment]
The latest version of both the Atlas and IGC connectors is 2.1:

- https://mvnrepository.com/artifact/org.odpi.egeria/egeria-connector-ibm-information-server
- https://mvnrepository.com/artifact/org.odpi.egeria/egeria-connector-hadoop-ecosystem

**Comment by planetf1** [comment]
Indeed - I realised that when writing the comment :-) 

Though we retrieve from jfrog rather than maven central.

The current code I'm testing sets them at 2.1 for now... enough to get them working  (hopefully)

**Comment by planetf1** [comment]
@cmgrote I note that in the apache atlas connector build currently we only have a simple jar

https://odpi.jfrog.io/odpi/egeria-release-local/org/odpi/egeria/egeria-connector-apache-atlas-adapter/2.1/

As per the pom, the connector is dependent on some atlas libraries in addition to core egeria dependencies -- so I suspect the connector will not load (the scripts currently look for the jar-with-dependencies). 

This probably means that as things stand the vdc environment will not work. The quickest fix is probably to go back to adding those dependencies, but it's an area we've thrashed around a bit on -- and as we speak the other connectors shipped in the egeria tree do not have dependencies. 

The information server package is currently building with jars.

So we need to either
 * Disable atlas by default / add notes that it no longer works (we have other issues with this connector)
 * Add back in the dependencies
 * consider another approach

I'm happy to make the change, but you know the connector repo the best. 

Need to decide if I get things working for the 2.2 release (delayed) or skip.

**Comment by cmgrote** [comment]
Full dependency connector jar is under this module:

https://odpi.jfrog.io/odpi/egeria-release-local/org/odpi/egeria/egeria-connector-hadoop-ecosystem-package/2.1/

**Comment by planetf1** [comment]
Fixed up, but then hit #3587. Going go go with a note in the release notes for 2.2 with a link to that issue reporting the atlas support as broken. (it's probably a build issue)

**Comment by planetf1** [comment]
I've done fixes to address this. We should revisit this when we return (if) the vdc chart. For now it will keep things up and running. So closing.
